packages/grafana-ui/src/components/FileDropzone/FileDropzone.tsx

```diff
@@ -0,0 +1,224 @@
+            });
+          };
+
+          reader.onprogress = (event) => {
+            setFileProperty(customFile, (fileToModify) => {
+              fileToModify.progress = event.loaded;
+            });
+          };
+
+          reader.onload = () => {
+            onLoad?.(reader.result);
+          };
+
+          reader.onerror = () => {
+            setFileProperty(customFile, (fileToModify) => {
+              fileToModify.error = reader.error;
+            });
+          };
+
+          read();
+        }
+      }
+    },
+    [onLoad, options, readAs, setFileProperty]
+  );
+
+  const removeFile = (file: DropzoneFile) => {
+    const newFiles = files.filter((f) => file.id !== f.id);
+    setFiles(newFiles);
+  };
+
+  const { getRootProps, getInputProps, isDragActive } = useDropzone({ ...options, onDrop });
+  const theme = useTheme2();
+  const styles = getStyles(theme, isDragActive);
+  const fileList = files.map((file) => <FileListItem key={file.id} file={file} removeFile={removeFile} />);
+
+  return (
+    <div className={styles.container}>
+      <div {...getRootProps({ className: styles.dropzone })}>
+        <input {...getInputProps()} />
+        {children ?? <FileDropzoneDefaultChildren primaryText={getPrimaryText(files, options)} />}
+      </div>
+      {options?.accept ? (
+        <small className={cx(styles.small, styles.acceptMargin)}>{getAcceptedFileTypeText(options)}</small>
+      ) : null}
+      {fileList}
+    </div>
+  );
+}
+
+export function FileDropzoneDefaultChildren({ primaryText = 'Upload file' }) {
+  const theme = useTheme2();
+  const styles = getStyles(theme);
+
+  return (
+    <div className={styles.iconWrapper}>
+      <Icon name="upload" size="xxl" />
+      <h3>{primaryText}</h3>
+      <small className={styles.small}>
+        Drag and drop here or <span className={styles.link}>Browse</span>
+      </small>
+    </div>
+  );
+}
+function getPrimaryText(files: DropzoneFile[], options?: DropzoneOptions) {
+  if (options?.multiple === undefined || options?.multiple) {
+    return 'Upload file';
+  }
+  return files.length ? 'Replace file' : 'Upload file';
+}
+
+function getAcceptedFileTypeText(options: DropzoneOptions) {
+  if (Array.isArray(options.accept)) {
+    return `Accepted file types: ${options.accept.join(', ')}`;
+  }
+
+  return `Accepted file type: ${options.accept}`;
+}
+
+function mapToCustomFile(file: File): DropzoneFile {
+  return {
+    id: uniqueId('file'),
+    file,
+    error: null,
+  };
+}
+
+function getStyles(theme: GrafanaTheme2, isDragActive?: boolean) {
+  return {
+    container: css`
+      display: flex;
+      flex-direction: column;
+      width: 100%;
+    `,
+    dropzone: css`
+      display: flex;
+      flex: 1;
+      flex-direction: column;
+      align-items: center;
+      padding: ${theme.spacing(6)};
+      border-radius: 2px;
+      border-width: 2px;
+      border-style: dashed;
+      border-color: ${theme.colors.border.medium};
```

Comment:
```suggestion
      border: 2px dashed ${theme.colors.border.medium};
```
This can be grouped to simplify the styles a bit.